File: CarePackage/Software/OperaGx.cs
namespace CarePackage.Software;

public class OperaGx : BaseSoftware
{
    public override string           Key            { get; set; } = "opera-gx";
    public override string           Name           { get; set; } = "Opera GX";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Browser;
    public override string           DownloadName   { get; set; } = "OperaGXSetup.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.operagx;
    public override string           Homepage       { get; set; } = "https://opera.com/gx";
    
    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://net.geo.opera.com/opera_gx/stable/windows");
}

File: packages/app/src/services/software/definitions/operaGx.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'opera-gx';
	public name = 'Opera GX';
	public category = [SoftwareCategory.Browser];
	public downloadName = 'OperaGXSetup.exe';
	public icon = 'operagx.png';
	public homepage = 'https://opera.com/gx';

	public async resolveDownloadUrl() {
		return ok('https://net.geo.opera.com/opera_gx/stable/windows');
	}
}
